.github/workflows/build-pr-check.yml

```diff
@@ -26,5 +29,25 @@ jobs:
         distribution: 'temurin'
         java-version: '11'
         cache: 'maven'
+    - name: Login to docker.io
+      if: github.event_name == 'pull_request'
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_PASSWORD }}
+        registry: docker.io
+    - name: Login to quay.io
+      if: github.event_name == 'pull_request'
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.QUAY_USERNAME }}
+        password: ${{ secrets.QUAY_PASSWORD }}
+        registry: quay.io
     - name: Build with Maven
       run: mvn -B clean install -U -Pintegration
+    - name: Build docker images
+      if: github.event_name == 'pull_request'
+      run: ./dockerfiles/build.sh --tag:${{ env.PR_IMAGE_TAG }}
+    - name: Push docker images
+      if: github.event_name == 'pull_request'
+      run: docker push quay.io/eclipse/che-server:${{ env.PR_IMAGE_TAG }}
```

Comment:
There will be dozen quay.io/eclipse/che-server:{pull_request.number} images if nobody clean up them.
Do you think there is a simple way to do it automatically?

Reply:
Yes, will be additional image which is related to every PR-number, but I think, it shouldn't reflect on quay.io repo. This image will be single for the PR, even if the PR will have many commits - the image just is rewritten. The same practise is already exist for 'che-dashboard' repository, you can take a look - https://quay.io/repository/eclipse/che-dashboard?tab=tags.
Also pay attention to quantity images which are related  with every commit to 'main' branch in the 'che-server' quay repository (https://quay.io/repository/eclipse/che-server?tab=tags), so I guess, an image with 'pr-number' tag won't affects the overall picture.

Reply:
There are 587 tags on https://quay.io/repository/eclipse/che-server?tab=tags and ~26 PR per month in this repo.
So, in a year you might expect + ~300 "pr-*" tags, which is a lot, IMHO.